src/utils/index.ts

```diff
@@ -8,3 +11,57 @@ export const patchPnpmEnv = (inputs: Inputs): NodeJS.ProcessEnv => ({
   ...process.env,
   PATH: getBinDest(inputs) + path.delimiter + process.env.PATH,
 })
+
+/**
+ * Reads and parses a potential .npmrc file to extract registry configuration
+ * @param workspaceRoot - The root directory of the workspace
+ * @returns The registry URL if found, otherwise undefined
+ */
+export async function readNpmrcRegistry(workspaceRoot: string): Promise<string | undefined> {
+  try {
+    const npmrcPath = path.join(workspaceRoot, '.npmrc');
+
+    debug(`Attempting to read .npmrc file from: ${npmrcPath}`);
+    const content = await readFile(npmrcPath, 'utf8');
+
+    for (const line of content.split('\n')) {
+      const trimmedLine = line.trim();
+
+      // skip comments and empty lines
+      if (!trimmedLine || /^[#;]/.test(trimmedLine)) {
+        continue;
+      }
+
+      const match = trimmedLine.match(/^registry\s*=\s*(.+)$/);
+      if (match) {
+        const registry = match[1].trim();
```

Comment:
The group `(.+)` should be a named group, e.g. `(?<url>.+)`.

`trimmedLine.match(regex)` should be replaced with `regex.exec(trimmedLine)`.

And `registry` can be extracted from `execResult.groups.registry`.

This whole thing can be a `.map(line => <regex>.exec(line).filter(Boolean)` or `.map(...).find(...)`. Whatever makes more sense.